internal/rbd/controllerserver.go

```diff
@@ -49,6 +49,9 @@ type ControllerServer struct {
 
 	// A map storing all volumes/snapshots with ongoing operations.
 	OperationLocks *util.OperationLock
+
+	// CR holds the credentials which is used to connect to the ceph cluster.
+	CR *util.Credentials
```

Comment:
should these credentials not be per volume? is it possible to have support for multiple ceph clusters in the same controller, similar to the csi provisioner/plugins?

Reply:
good catch! Yes, correct. this cannot be kept in ControllerServer. added new logic for this one.